src/rpc/blockchain.cpp

```diff
@@ -552,15 +570,40 @@ static RPCHelpMan getblockheader()
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Block not found");
     }
 
-    if (!fVerbose)
+    if (count == 0)
```

Comment:
I don't think it's a good idea to duplicate the logic for each "mode".

Better to build a 1-item array, and return only `headers[0]` as a special case.